lib/report/generate.js

```diff
@@ -0,0 +1,159 @@
+#! /usr/bin/env node
+
+// Functions and behavior for compiling the report.
+
+// RESOURCES:
+//   - https://www.npmjs.com/package/commander
+//   - https://developer.atlassian.com/blog/2015/11/scripting-with-node/
+//   - https://github.com/tj/commander.js/tree/master/examples
+//   - https://nodejs.org/api/fs.html
+//   - https://nodejs.org/api/path.html
+//   - https://github.com/paulmillr/chokidar
+
+var path = require('path');
+var chokidar = require('chokidar');
+var _ = require('lodash');
+var version = require("../version").version;
+var fileUtils = require("../file-utils");
+var settings = require("../settings");
+
```

Comment:
Below are some helper functions I used to convert the report generator to support plugins.